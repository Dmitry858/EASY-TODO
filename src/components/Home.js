import React from 'react';
import { connect } from 'react-redux';

const Home = (props) => {

    return (
        <div className="content">
            <div className="container">
                <p>Далеко-далеко за словесными горами в стране гласных и согласных живут рыбные тексты. Деревни последний грамматики языком скатился большого! Пустился жаренные всеми снова семантика маленькая ipsum залетают прямо последний. Пустился от всех не проектах! Имени не необходимыми lorem, свой пояс сбить реторический все! Обеспечивает своего назад живет одна не речью рукописи, диких текста повстречался, лучше прямо текст текстов курсивных безопасную, предупреждал раз! Напоивший, единственное.</p>
                <p>Предупреждал семь по всей, о букв страну ее предупредила буквенных жизни путь первую текстами раз, за себя ipsum сбить семантика рыбного взгляд текст использовало! Языком образ, правилами заголовок не над однажды. Необходимыми текста сих там, не пояс буквенных парадигматическая? Путь не продолжил безорфографичный инициал необходимыми ему, сих толку бросил своего, грамматики он силуэт речью над живет ipsum родного ведущими даль там?</p>
                <p>Напоивший безорфографичный даль, раз букв ipsum семь вдали ведущими моей океана свой заголовок диких образ рыбного грамматики. Вопрос эта ее свою власти о страну. Деревни мир коварный маленькая безопасную сбить.</p>
                <p>Далеко-далеко за словесными горами в стране гласных и согласных живут рыбные тексты. Деревни последний грамматики языком скатился большого! Пустился жаренные всеми снова семантика маленькая ipsum залетают прямо последний. Пустился от всех не проектах! Имени не необходимыми lorem, свой пояс сбить реторический все! Обеспечивает своего назад живет одна не речью рукописи, диких текста повстречался, лучше прямо текст текстов курсивных безопасную, предупреждал раз! Напоивший, единственное.</p>
                <p>Предупреждал семь по всей, о букв страну ее предупредила буквенных жизни путь первую текстами раз, за себя ipsum сбить семантика рыбного взгляд текст использовало! Языком образ, правилами заголовок не над однажды. Необходимыми текста сих там, не пояс буквенных парадигматическая? Путь не продолжил безорфографичный инициал необходимыми ему, сих толку бросил своего, грамматики он силуэт речью над живет ipsum родного ведущими даль там?</p>
                <p>Напоивший безорфографичный даль, раз букв ipsum семь вдали ведущими моей океана свой заголовок диких образ рыбного грамматики. Вопрос эта ее свою власти о страну. Деревни мир коварный маленькая безопасную сбить.</p>
                <p>Далеко-далеко за словесными горами в стране гласных и согласных живут рыбные тексты. Деревни последний грамматики языком скатился большого! Пустился жаренные всеми снова семантика маленькая ipsum залетают прямо последний. Пустился от всех не проектах! Имени не необходимыми lorem, свой пояс сбить реторический все! Обеспечивает своего назад живет одна не речью рукописи, диких текста повстречался, лучше прямо текст текстов курсивных безопасную, предупреждал раз! Напоивший, единственное.</p>
                <p>Предупреждал семь по всей, о букв страну ее предупредила буквенных жизни путь первую текстами раз, за себя ipsum сбить семантика рыбного взгляд текст использовало! Языком образ, правилами заголовок не над однажды. Необходимыми текста сих там, не пояс буквенных парадигматическая? Путь не продолжил безорфографичный инициал необходимыми ему, сих толку бросил своего, грамматики он силуэт речью над живет ipsum родного ведущими даль там?</p>
                <p>Напоивший безорфографичный даль, раз букв ipsum семь вдали ведущими моей океана свой заголовок диких образ рыбного грамматики. Вопрос эта ее свою власти о страну. Деревни мир коварный маленькая безопасную сбить.</p>
            </div>
        </div>
    );

};

export default connect(state => state)(Home);