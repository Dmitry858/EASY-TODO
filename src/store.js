import { createStore, combineReducers, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import tasksStoreConfig from './reducers/tasks';

const createReducer = (initialState, handlers) => {
    return (state = initialState, action) => {
        return (handlers[action.type] && handlers[action.type](state, action)) || state;
    };
};

const tasksReducers = createReducer(tasksStoreConfig.initialState, tasksStoreConfig.actions);

const rootReducer = combineReducers({
    tasks: tasksReducers,
});

export default createStore(rootReducer, {}, applyMiddleware(thunk));