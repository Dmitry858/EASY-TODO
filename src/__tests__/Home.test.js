import React from 'react';
import ReactDOM from 'react-dom';
import { Router } from 'react-router-dom';
import { createBrowserHistory } from 'history';
import qhistory from 'qhistory';
import { stringify, parse } from 'qs';
import Home from '../components/Home';
import { Provider } from 'react-redux';
import store from '../store.js';
import { render } from '@testing-library/react';

describe('Home testing', () => {
    const history = qhistory(
        createBrowserHistory({}),
        stringify,
        parse
    );

    const HomeWithProvider = () => {
        return (
            <Provider store={store}>
                <Router history={history}>
                    <Home />
                </Router>
            </Provider>
        );      
    };
    
    it('Home render without crashing', () => {
        const div = document.createElement('div');
      
        ReactDOM.render(
            <HomeWithProvider />,
            div
        );
        ReactDOM.unmountComponentAtNode(div);
    });

    it('Home matches the snapshot', () => {
        const { container } = render(<HomeWithProvider />);
        expect(container).toMatchSnapshot();
    });
    
});
