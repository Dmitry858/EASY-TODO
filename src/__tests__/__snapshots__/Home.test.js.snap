// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home testing Home matches the snapshot 1`] = `
<div>
  <div
    class="content"
  >
    <div
      class="container"
    >
      <div
        class="get-started row valign-wrapper"
      >
        <div
          class="col m6 center-align"
        >
          <h1
            class="center-align"
          >
            Приведите дела 
            <br />
            в порядок с 
            <br />
            <span>
              EASY TODO
            </span>
          </h1>
          <a
            class="btn btn-large modal-trigger waves-effect"
            href="#modal-signup"
          >
            Начать
          </a>
        </div>
        <div
          class="col m6 center-align"
        >
          <img
            alt="Приведите дела в порядок"
            src="image1.png"
          />
        </div>
      </div>
      <div
        class="screenshot"
      >
        <img
          alt="Приложение EASY TODO"
          src="screen.jpg"
        />
      </div>
    </div>
    <div
      class="read-more"
    >
      <div
        class="container center-align"
      >
        <p>
          Позвольте себе не думать постоянно о ежедневных заботах, переместите свои дела из головы в планировщик задач и обретите спокойствие.
        </p>
        <a
          href="/about"
        >
          <i
            aria-hidden="true"
            class="fa fa-arrow-right"
          />
          Узнать подробнее о приложении
        </a>
      </div>
    </div>
    <div
      class="container"
    >
      <div
        class="relax row valign-wrapper"
      >
        <div
          class="col m6 center-align"
        >
          <img
            alt="Приведите дела в порядок"
            src="image2.png"
          />
        </div>
        <div
          class="col m6 center-align"
        >
          <h3
            class="center-align"
          >
            EASY TODO поможет организовать задачи наилучшим образом, чтобы у вас освободилось больше времени для отдыха
          </h3>
          <a
            class="btn btn-large modal-trigger waves-effect"
            href="#modal-signup"
          >
            Начать
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
